runtime/src/accounts.rs

```diff
@@ -118,7 +119,12 @@ impl Accounts {
                 .filter(|key| !message.program_ids().contains(&key))
             {
                 let (account, rent) = AccountsDB::load(storage, ancestors, accounts_index, key)
-                    .and_then(|(account, _)| rent_collector.update(account))
+                    .and_then(
+                        |(mut account, _)| match rent_collector.update(&mut account) {
```

Comment:
I think rent_collector can do this work for us.

What if it returned Account::default() instead of None?

Reply:
@rob-solana But, then how would caller differentiate between two cases? This limits caller's visibility. (Might be useful in test and  in certain scenarios where we want to fail instead of just storing default).

Reply:
storing default() is what this code did before rent, @sakridge has a solution

Reply:
What I meant is, it is not good idea to design a function that doesn't indicate what it did with the mutable account. Did it overwrite it or not overwrite it? Basically limits the visibility of caller.

What we should do is add a boolean alongside that indicates whether the returned account is default or original account. 

What do you think?

Reply:
@rob-solana On hindsight, without passing Account by value this is not possible, as `Account::default()` will be created inside `RentCollector.update`, so we cannot return a reference to it. 

Current design is most idiomatic one.

Reply:
passing by value would be bad, I think

maybe a boolean would look nicer

Reply:
actually, what if you just filled the ref Account with Account::default() values?

Reply:
@rob-solana implemented something similar but more elegant. Check it out.